#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_fix_gpgme.dpatch by Jan Wagner <waja@cyconet.org>
##
## DP: check gpgme version

@DPATCH@
diff -urNad openvas-libnasl-2.0.2~/nasl/nasl_signature.c openvas-libnasl-2.0.2/nasl/nasl_signature.c
--- openvas-libnasl-2.0.2~/nasl/nasl_signature.c	2009-08-18 10:06:59.000000000 +0200
+++ openvas-libnasl-2.0.2/nasl/nasl_signature.c	2009-09-17 09:55:02.000000000 +0200
@@ -171,7 +171,12 @@
   gpgme_ctx_t ctx = NULL;
   char * gpghome = determine_gpghome();
 
+  /* Calls seem to be necessary for certain versions of gpgme (for
+    initialization). Note that we could check the version number here, but do so
+    in configure. */
+  gpgme_check_version (NULL);
   err = gpgme_engine_check_version(GPGME_PROTOCOL_OpenPGP);
+
   if (err)
     {
       print_gpgme_error("gpgme_engine_check_version", err);
